### DISCUSSION TITLE
gatsby-cli: gatsby new peerDependency issue

### DISCUSSION THREAD

**Discussion by enyelsequeira** [gatsbyjs/gatsby#30708] [Help]
## Description

Whenever running the gatsby new in China it does not work, tried with VPN and without it. it starts loading then it gives an error

### Steps to reproduce

run gatsby new 
go through the process
and start installation

### Expected result

new project should be created without any errors 

### Actual result

errors when creating error below

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: webpack@5.28.0
npm ERR! node_modules/webpack
npm ERR!   webpack@"^5.16.0" from gatsby@3.1.2
npm ERR!   node_modules/gatsby
npm ERR!     gatsby@"^3.1.2" from the root project
npm ERR!   peer webpack@">=4.43.0 <6.0.0" from @pmmmwh/react-refresh-webpack-plugin@0.4.3
npm ERR!   node_modules/@pmmmwh/react-refresh-webpack-plugin
npm ERR!     @pmmmwh/react-refresh-webpack-plugin@"^0.4.3" from gatsby@3.1.2
npm ERR!     node_modules/gatsby
npm ERR!       gatsby@"^3.1.2" from the root project
npm ERR!   1 more (webpack-dev-server)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer webpack@"^4.20.0" from webpack-plugin-ramdisk@0.1.8
npm ERR! node_modules/webpack-plugin-serve/node_modules/webpack-plugin-ramdisk
npm ERR!   webpack-plugin-ramdisk@"^0.1.2" from webpack-plugin-serve@1.3.0
npm ERR!   node_modules/webpack-plugin-serve
npm ERR!     peer webpack-plugin-serve@"0.x || 1.x" from @pmmmwh/react-refresh-webpack-plugin@0.4.3
npm ERR!     node_modules/@pmmmwh/react-refresh-webpack-plugin
npm ERR!       @pmmmwh/react-refresh-webpack-plugin@"^0.4.3" from gatsby@3.1.2
npm ERR!       node_modules/gatsby
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/enyel/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:

 ERROR

Command failed with exit code 1: npm install --loglevel error --color always


```
### Environment

 System:
    OS: macOS 11.2.3
    CPU: (8) x64 Apple M1
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.16.0 - ~/.volta/tools/image/node/14.16.0/bin/node
    Yarn: 1.22.10 - ~/.volta/tools/image/yarn/1.22.10/bin/yarn
    npm: 7.6.1 - ~/.volta/tools/image/npm/7.6.1/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 89.0.4389.90
    Firefox: 86.0.1
    Safari: 14.0.3

**Comment by LekoArts** [comment]
`gatsby new` is running a `npm install` internally and unfortunately npm 7 broke compatibility with some peerDependency resolving - that's why it shows you the message to use the `--legacy-peer-deps` flag.

I see that you also have `yarn` installed, you could switch your default package manager to yarn to work around this issue:
https://www.gatsbyjs.com/docs/reference/gatsby-cli/#how-to-change-your-default-package-manager-for-your-next-project
